Automattic/jetpack
# Increase timeout of license activation request from 10 to 30 seconds

Asana card: 1201096622142517-as-1201400491645539

#### Changes proposed in this Pull Request:

* Increase the timeout of the request that triggers the activation of a product via license key. This is particularly important for Jetpack Complete, which activation requires some extra work that is not always completed in under 10 seconds (you can read more about this in [here](p1637846404079400-slack-C82FZ5T4G) and [here](1201096622142517-as-1201400491645539)).

**Note**: this is a workaround until we find a proper solution on the WPCOM that decreases the activation time for Jetpack Complete (which might not happen before the 10.4 gets released).

#### Jetpack product discussion



* Project Thread: p1HpG7-cSg-p2.
* This is a known issue described in p1HpG7-dIr-p2.

#### Does this pull request change what data or activity we track or use?


* Not at all.

#### Testing instructions:






* Spin up a site using this Jetpack branch.
* Connect Jetpack to your A8C WordPress.com account.
* Enable the A8c Proxy.
* Go to https://cloud.jetpack.com/pricing.
* Select Jetpack Complete.
* Complete checkout.
* On the "Thank you for purchase!" page, just stop there.. (Don't enter a site URL in the input box).
* On your site, visit `/wp-admin/admin.php?page=jetpack#/license/activation`.
* Open the Network section of the JS Console.
* Enter the license key associated with your Jetpack Complete purchase (you can get it from SA).
* Activate the product.
* Once the activation finishes, do not refresh the page.
* Verify how long did the `attach_licenses` request take to complete. If it took more than 10 seconds, that means that the timeout was effectively increased.
* Click on Jetpack > Dashboard (sidebar).
* Click on the My Plan tab.
* Verify that you see Jetpack Complete in the My Plan section.
* Verify that you see `Instant search and filtering` card just below that.
* Go to Jetpack > Settings (sidebar).
* Click on the Performace tab.
* Verify that Search and Instant Search Experience are enabled.

<img width="967" alt="Screen Shot 2021-11-30 at 10 42 29" src="https://user-images.githubusercontent.com/3418513/144058692-b2bc2444-b5c9-4c67-a880-cf004a80e99a.png">
